test/parallel_api/algorithm/alg.sorting/alg.binary.search/binary.search/binary_search.pass.cpp

```diff
@@ -17,107 +17,153 @@
+#    endif
+        host_first = get_host_pointer(first);
+        host_val_first = get_host_pointer(value_first);
+        host_result = get_host_pointer(result_first);
+        check_values(host_result, n);
+    }
 #endif
 
-bool test_on_host()
-{
-    int key[10] = {0, 2, 2, 2, 3, 3, 3, 3, 6, 6};
-    int val[5] = {0, 2, 4, 7, 6};
-    int res[5];
-  
-     // call algorithm
-     oneapi::dpl::binary_search(oneapi::dpl::execution::par, std::begin(key), std::end(key), std::begin(val), std::end(val), std::begin(res), std::less<int>());
-
-     //check data
-     if((res[0] != true) || (res[1] != true) || (res[2] !=false) || (res[3] != false) || (res[4] != true))
-         std::cout << "binary_search on host FAIL." << std::endl;
-
-     return 0;
-}
+    // specialization for host execution policies
+    template <typename Policy, typename Iterator1, typename Iterator2, typename Iterator3, typename Size>
+    typename ::std::enable_if<
+#if TEST_DPCPP_BACKEND_PRESENT
+        !oneapi::dpl::__internal::__is_hetero_execution_policy<typename ::std::decay<Policy>::type>::value &&
+#endif
+            !is_same_iterator_category<Iterator3, ::std::bidirectional_iterator_tag>::value &&
+            !is_same_iterator_category<Iterator3, ::std::forward_iterator_tag>::value,
+        void>::type
+    operator()(Policy&& exec, Iterator1 first, Iterator1 last, Iterator2 value_first, Iterator2 value_last,
+               Iterator3 result_first, Iterator3 result_last, Size n)
+    {
+        typedef typename ::std::iterator_traits<Iterator1>::value_type ValueT;
+
+        // call algorithm with no optional arguments
+        initialize_data(first, value_first, result_first, n);
 
-int main()
+        auto res1 = oneapi::dpl::binary_search(exec, first, last, value_first, value_last, result_first);
+        check_values(result_first, n);
+
+        // call algorithm with comparator
+        initialize_data(first, value_first, result_first, n);
+
+        auto res2 =
+            oneapi::dpl::binary_search(exec, first, last, value_first, value_last, result_first, ::std::less<ValueT>());
+        check_values(result_first, n);
+    }
+
+    // specialization for non-random_access iterators
+    template <typename Policy, typename Iterator1, typename Iterator2, typename Iterator3, typename Size>
+    typename ::std::enable_if<is_same_iterator_category<Iterator3, ::std::bidirectional_iterator_tag>::value ||
+                                  is_same_iterator_category<Iterator3, ::std::forward_iterator_tag>::value,
+                              void>::type
+    operator()(Policy&& exec, Iterator1 first, Iterator1 last, Iterator2 value_first, Iterator2 value_last,
+               Iterator3 result_first, Iterator3 result_last, Size n)
+    {
+    }
+};
+
+int
+main()
 {
 #if TEST_DPCPP_BACKEND_PRESENT
-    test_on_device();
+    test3buffers<uint64_t, test_binary_search>();
 #endif
-    test_on_host();
-
     return TestUtils::done();
```

Comment:
Am I right that tests should be skipped if `TEST_DPCPP_BACKEND_PRESENT==0`?

Reply:
Tests on host are reintroduced now, so only the tests on device should be skipped when `TEST_DPCPP_BACKEND_PRESENT==0`